content/tutorials/manage-the-rippled-server/configuration/run-rippled-as-a-validator.md

```diff
@@ -1,74 +1,191 @@
+To help protect a validator from [DDoS](https://en.wikipedia.org/wiki/Denial-of-service_attack) attacks, set up one or more stock `rippled` servers as proxies between the validator and inbound traffic. ***TODO: Stated correctly? I think inbound traffic is clearer than "outside network" - but is it accurate? Do we have any recommendations around the minimum number of proxies to set up? Is the primary purpose of providing more than one proxy that if one goes down, you don't lose your connection to the network? Also curious about how the validator's "view" of the network changes when proxies are in place -- what does inbound traffic to a proxy-less validator look like vs inbound traffic to a validator with two proxies, for example? I don't quite understand how the views could differ, if at all.***
 
-5. Configure the validator host machine's firewall to only accept inbound connections from its public-facing rippled(s).
+1. [Enable validation](#enable-validation-on-your-rippled-server) on your `rippled` server.
 
-Remember to restart `rippled` for config changes to take effect.
+2. Set up stock `rippled` servers. For more information, see [Install rippled](install-rippled.html).
 
-Take care not to publish the IP address of your validator.
+3. Configure your validator and stock `rippled` servers to run in a [cluster](cluster-rippled-servers.html).
 
+4. In your validator's `rippled.cfg` file:
 
-## Domain Verification
+    1. Copy the `[ips_fixed]` IP address list you defined in the previous cluster configuration step and paste it under `[ips]`.
 
-Network participants are unlikely to trust validators without knowing who is operating them. To address this concern, validator operators can associate their validator with a web domain that they control.
+        The `[ips_fixed]` and `[ips]` values must be identical and contain only the IP addresses and ports of the stock `rippled` servers you clustered with your validator. This ensures that your validator connects to only the stock `rippled` servers that you control.
 
-1. Find your validator public key by running the following on the validator server:
+    2. Set `[peer_private]` to `1` to prevent your validator's IP address from being forwarded. For more information, see [Private Peers](peer-protocol.html#private-peers).
 
+        **Warning:** Be sure that you don't publish your validator's IP address via any other channels as well.
+
+5. Configure your validator host machine's firewall to accept inbound connections from only the stock `rippled` servers you clustered it with.
+
+6. Restart `rippled`.
+
+        $ sudo systemctl restart rippled.service
+
+***TODO: should we provide some tips on how to test that the firewall has been configured correctly? maybe ping the validator IP address from a machine outside the cluster? The test that the cluster is set up correctly (peers command) is covered in the clustering doc.***
```

Comment:
Thank you for this. Yes, the clustering doc provides the check using the `peers` method. I've added the test using the Peer Crawler Endpoint - 👌 !

Can I just verify the following with you?

The validator (and its IP address) is not included in the Peer Crawler endpoint response from the stock server in the cluster. Is this because [peer_private] is set to `1` in the validator's `rippled.cfg`? 

Though, when I run `peers` on the same stock server in the cluster, the validator does appear with its IP address in the results. This is okay because I can run `peers` only on servers I have admin control over and because of the way my validator's [ips] and [ips_fixed] values are set - I know that my validator will not connect to any other peers than the ones I control in the cluster.

Reply:
Correct.

One thing that may be confusing is that enabling `[peer_private]` does several things:

- Prevents the server from opening peer connections to unknown servers. (You have to explicitly configure which servers you want to peer with.)
- Makes the server reject peer connections from unknown servers. (This is an extra defense alongside the firewall you're expected to use.)
- Requests that the server's peers don't mention it in their peer crawler results.

The `peers` command is an admin command, so it provides an accurate picture of which servers are peered with the server you query. That's important for administering the server!

The peer crawler's results are treated as public. You can get peer crawler results without being an administrator. Honest servers (that is, ones that haven't been modified to break this rule) won't report their private peers in the peer crawler results. You could modify your server so that it _would_ report its private peers, but they can only spill their own secrets (that is, private peers they're _directly_ connected with) since they don't know anyone else's secrets.